pkg/cloudprovider/servicelb.go

```diff
@@ -174,6 +176,13 @@ func (k *k3s) onChangeEndpointSlice(key string, eps *discovery.EndpointSlice) (*
 	return eps, nil
 }
 
+// onChangeService handles changes to Services. This is used to ensure that changes in the Service are
+// reflected in the LoadBalancer status (e.g. changing to externalTrafficPolicy=Local which should remove externalIPs)
+func (k *k3s) onChangeService(key string, svc *core.Service) (*core.Service, error) {
```

Comment:
I'm curious why this is necessary. This would appear to duplicate the OnChange handler in the core Kubernetes cloud-controller code base, which calls EnsureLoadBalancer or UpdateLoadBalancer when the Service object changes. Is it not calling those handlers on the right events to see these changes?

Reply:
It's not super expensive but it does mean that both the built-in CCM and Wrangler are now watching Services. Adding a dummy annotation so that the behavior of the daemonset stays consistent might be more efficient than adding another watch?

Reply:
I'll add the label to the ds. That way, this new alternative, is more similar to what we are currently doing